pkgs/by-name/tr/tractor/package.nix

```diff
@@ -0,0 +1,80 @@
+{
+  lib,
+  fetchFromGitLab,
+  glib,
+  python3Packages,
+  gobject-introspection,
+  gsettings-desktop-schemas,
+  tor,
+  obfs4,
+  snowflake,
+  wrapGAppsHook4,
+}:
+
+let
+  # This package should be updated together with pkgs/by-name/ca/carburetor/package.nix
+  version = "5.0.0";
+in
+python3Packages.buildPythonApplication {
+  pname = "tractor";
+  inherit version;
+
+  pyproject = true;
+
+  src = fetchFromGitLab {
+    domain = "framagit.org";
+    owner = "tractor";
+    repo = "tractor";
+    rev = version;
+    hash = "sha256-KyVL3dFofoi2TRtZo557X9P/RD16v94VuWdtdAskZk4=";
+  };
+
+  patches = [ ./fix-gsettings-schema.patch ];
+
+  nativeBuildInputs = [
+    glib
+    gobject-introspection
+    gsettings-desktop-schemas
+    wrapGAppsHook4
+  ];
+
+  propagatedBuildInputs = [
+    tor
+    obfs4
+    snowflake
+  ];
+
+  dependencies = [
+    python3Packages.setuptools
+    python3Packages.fire
+    python3Packages.pygobject3
+    python3Packages.pysocks
+    python3Packages.stem
+  ];
+
+  postInstall = ''
+    mkdir -p "$out/share/glib-2.0/schemas"
+    cp "$src/src/tractor/tractor.gschema.xml" "$out/share/glib-2.0/schemas"
+
+    substituteInPlace "$out/share/glib-2.0/schemas/tractor.gschema.xml" --replace-fail '/usr/bin/obfs4proxy' '${obfs4}/bin/lyrebird'
+    substituteInPlace "$out/share/glib-2.0/schemas/tractor.gschema.xml" --replace-fail '/usr/bin/snowflake-client' '${snowflake}/bin/client'
+    substituteInPlace "$out/share/glib-2.0/schemas/tractor.gschema.xml" --replace-fail '/usr/bin/conjure-client' 'none'
```

Comment:
I added these three subs to replace the default paths with the nix store ones.
The third one is unnecessary. should I remove it?

also, should we make them optional?

Reply:
Oh, I found conjure

```
$ nix-locate conjure-client
/nix/store/f3k3jx8nfnkqjdb022rgq2cy0yx1yzpi-tor-browser-14.0.3/share/tor-browser/TorBrowser/Tor/PluggableTransports/conjure-client
```

Reply:
Though cannot test it because

```
Dec 22 18:38:59.000 [notice] Managed proxy "/nix/store/f3k3jx8nfnkqjdb022rgq2cy0yx1yzpi-tor-browser-14.0.3/
share/tor-browser/TorBrowser/Tor/PluggableTransports/conjure-client": retrying conjure registration, station is
under high load.
```

Reply:
> Oh, I found conjure
> 
> ```
> $ nix-locate conjure-client
> /nix/store/f3k3jx8nfnkqjdb022rgq2cy0yx1yzpi-tor-browser-14.0.3/share/tor-browser/TorBrowser/Tor/PluggableTransports/conjure-client
> ```

I didn't know about nix-locate! cool!

Should we make this a dependency of the tor browser? 
I think it might be better to package the transport plugins separately. 🤔

Reply:
> I think it might be better to package the transport plugins separately

Would be great to package conjure separately, but adding `withConjure ? false` is good enough for now

Reply:
> > I think it might be better to package the transport plugins separately
> 
> Would be great to package conjure separately, but adding `withConjure ? false` is good enough for now

Are there any examples for using `lib.optional` in the scripts? I want to skip the string replaces.